### PR TITLE
Break out release notes from user guide in their own page

### PR DESCRIPTION
**Pull request by gaganis**
Issue: #1068

---

I hereby agree to the terms of the JUnit Contributor License Agreement.

### CODE REVIEW

**Comment by sbrannen** [comment]
Thanks for the PR! 👍 

We'll have a look at it.

**Comment by sbrannen** [comment]
Specifically, we have to convert the internal cross-reference links to “Inter-document cross references” as explained in the documention link provided above by @sormuras.

**Comment by gaganis** [comment]
Thanks @sormuras @sbrannen for the comments and tips. Will look into it on the weekend :)

File: documentation/src/docs/asciidoc/release-notes-5.0.0-M1.adoc [hunk 1]
```diff
@@ -23,7 +23,7 @@ on GitHub.
 ** See the https://github.com/junit-team/junit5/labels/up-for-grabs[up-for-grabs] label
    on GitHub.
 * Group IDs, artifact IDs, and versions have changed for all published artifacts.
-** See <<release-notes-5.0.0-m1-migration-artifacts>> and <<dependency-metadata>>.
+** See <<release-notes-5.0.0-m1-migration-artifacts>> and <<index.adoc#dependency-metadata,Dependency Metadata>>.
```
Comment (sbrannen) [inline documentation/src/docs/asciidoc/release-notes-5.0.0-M1.adoc]: Does the cross reference support not pick up the "Dependency Metadata" section header automatically?

File: documentation/src/docs/asciidoc/index.adoc [hunk 2]
```diff
@@ -102,4 +31,8 @@ include::api-evolution.adoc[]
 
 include::contributors.adoc[]
 
-include::release-notes.adoc[]
+
+[[release-notes]]
+== Release Notes
+
+Release notes have been moved link:release-notes.html[here]
```
Comment (sbrannen) [inline documentation/src/docs/asciidoc/index.adoc]: Please use a cross reference here instead of the external link format.

Comment (sbrannen) [reply documentation/src/docs/asciidoc/index.adoc]: Please change the text to:

> The release notes are available _here_.

File: documentation/src/docs/asciidoc/release-notes-5.0.0-M1.adoc (hunk 1 above)
Comment (gaganis) [reply documentation/src/docs/asciidoc/release-notes-5.0.0-M1.adoc]: Unfortunately asciidoctor `cannot(yet) resolve titles automatically`. Please see: http://asciidoctor.org/docs/user-manual/#inter-document-cross-references

Comment (sbrannen) [reply documentation/src/docs/asciidoc/release-notes-5.0.0-M1.adoc]: Ahhhh.... too bad!

I missed that nuance.

**Comment by sbrannen** [comment]
This has been merged into `master`.

Thanks

**Pull request merged**